evm/src/executor/inspector/cheatcodes/ext.rs

```diff
@@ -18,19 +18,44 @@ use std::{
     process::Command,
     str::FromStr,
 };
+use tracing::trace;
 
+/// Invokes a `Command` with the given args and returns the abi encoded response
+///
+/// If the output of the command is valid hex, it returns the hex decoded value
 fn ffi(state: &Cheatcodes, args: &[String]) -> Result<Bytes, Bytes> {
-    let output = Command::new(&args[0])
+    if args.is_empty() || args[0].is_empty() {
+        return Err(util::encode_error("Can't execute empty command"))
+    }
+    let mut cmd = Command::new(&args[0]);
+    if args.len() > 1 {
+        cmd.args(&args[1..]);
+    }
+
+    trace!(?args, "invoking ffi");
+
+    let output = cmd
         .current_dir(&state.config.root)
-        .args(&args[1..])
         .output()
-        .map_err(util::encode_error)?
-        .stdout;
-    let output = unsafe { std::str::from_utf8_unchecked(&output) };
-    let decoded = hex::decode(&output.trim().strip_prefix("0x").unwrap_or(output))
-        .map_err(util::encode_error)?;
+        .map_err(|err| util::encode_error(format!("Failed to execute command: {}", err)))?;
+
+    if state.config.verbosity >= 3 {
```

Comment:
Not sure how I feel about this since it's w/o context - tests are run in parallel, so this is potentially going to be printed next to something entirely unrelated

Talked about maybe adding a `CheatcodeFailed(string msg, string[] context)` error type or something we could revert with instead to solve this, but that's obviously a bigger piece of work. Unsure if you want to wait with this stderr thing until then? Also removes the need to pass `verbosity` to the cheatcode inspector.

I also wrote about something similar a long time ago here: https://github.com/foundry-rs/foundry/issues/928

Reply:
Agree with bjerg, would recommend we skip that change here and think about it a bit more in #928